book/todo.md

```diff
@@ -0,0 +1,87 @@
+# TODO
+
+## Concepts
+- Big Picture Architecture: A low-level framework. Threads (actors) communicating async via channels. No global constructs in Hydroflow. No SPMD assumptions.
+- Hydroflow and Rust: how do they go together?
+    - State, control, scoping
+- Time
+    - The Hydroflow event loop and Ticks. local clock
+    - Assembling a global vector clock
+- Fix points and Strata
+- State over time
+    - lifetimes
+    - explicit deletion
+- Coordination tricks?
+    - End-of-stream to Distributed EOS?
+
+## Docs
+- `Hydroflow` trait and its methods
+- Review the ops docs
+
+## What's covered in examples
+- Concepts covered
+    - cargo generate for templating
+    - Hydroflow program specs embedded in Rust
+    - Tokio Channels and how to use them in Hydroflow
+        - Network sources and sinks (source_stream)
+        - Built-in serde (source_stream_serde, dest_sink_serde)
+    - Hydroflow syntax: operators, ->, variables, indexing multi-input/output operators
+    - running Hydroflow via `run_available` and `run_async`
+    - Recursion via cyclic dataflow
+    - Fixpoints and Strata
+    - Template structure: clap, message types
+    - source_stdin
+    - Messages and `demux`
+    - broadcast pattern
+    - gated buffer pattern
+    - bootstrapping pipelines
+
+- Operators covered
+    - cross_join
+    - demux
+    - dest_sink_serde
+    - difference
+    - filter
+    - filter_map
+    - flatten
+    - flat_map
+    - for_each
+    - join
+    - map
+    - merge
+    - source_iter
+    - source_stdin
+    - source_stream
+    - source_stream_serde
+    - tee
+    - unique
+
+- Operators not discussed
+    - dest_sink
+    - identity
+    - repeat_iter
+    - unzip
+    - *fold* -- add to chapter on [state](state.md)
+    - *reduce* -- add to chapter on [state](state.md)
+    - *group_by* -- add to chapter on [state](state.md)
+    - *sort* -- add to chapter on [state](state.md)
+    - *sort_by* -- add to chapter on [state](state.md)
+    - *next_stratum* -- add to chapter on [time](time.md)
+    - *next_epoch* -- add to chapter on [time](time.md)
+    - *inspect* -- add to chapter on [debugging](debugging.md)
+    - *null* -- add to chapter on [debugging](debugging.md)
+
+## How-Tos
+- A partitioned Service
+- A replicated Service
+- Interfacing with external data
+- Interfacing with external services
+
+## Odds and ends taken out of other chapters
+- **Document the methods on the `hydroflow` trait** -- especially the run methods.
```

Comment:
> `hydroflow` struct

instead of trait

Reply:
Can note that I need to document "core API" usage i.e. using the `hydroflow` struct directly without surface syntax